negotiation/neural/batcher.py

```diff
@@ -28,8 +31,9 @@ def __init__(self, encoder_args, decoder_args, context_data, vocab,
         if sort_by_length:
             for k, v in self.context_data.iteritems():
                 self.context_data[k] = self.order_by_id(v, sorted_ids)
-            for attr in ('encoder_inputs', 'decoder_inputs', 'targets', 'lengths'):
-                setattr(self, attr, self.order_by_id(getattr(self, attr), sorted_ids))
+            for attr in ('encoder_inputs', 'decoder_inputs', 'item_inputs', 'context_inputs', 'targets', 'lengths'):
+                sorted_attrs = self.order_by_id(getattr(self, attr), sorted_ids)
+                setattr(self, attr, sorted_attrs)
 
         if time_major:
             for attr in ('encoder_inputs', 'decoder_inputs', 'targets'):
```

Comment:
make context inputs time_major too. so it is processed in the same way as encoder inputs later in the code.